### PR TITLE
fix: add missing typing for [no-any-return] (TDE-267) 

### PR DESCRIPTION
**Pull request by paulfouquet**
### Change Description:
Fix where return a value with an Any type in a function that is annotated to return a non-Any value.

### CODE REVIEW

**Review by amfage — APPROVED** [review]

File: topo_processor/stac/asset.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def get_content_type(self) -> str:
             return self.content_type
         return MimeTypes().guess_type(self.target if self.target else self.source_path)[0]
 
-    def get_checksum(self) -> str:
+    def get_checksum(self) -> Any:
```
Comment (MDavidson17) [inline topo_processor/stac/asset.py:44]: So, there seems to be three options we could pick from and mypy passes?
Is there a rule or best practise for how we should choose?
1. `-> Any`
2. `-> Union[str, Any]`
3. `-> str...
     ... return cast(str, value)`

**Review by MDavidson17 — APPROVED** [review]
Just the one comment which is more a question, otherwise looks good to me.

File: topo_processor/stac/asset.py (hunk 1 above)
Comment (paulfouquet) [reply topo_processor/stac/asset.py:44]: Good spot. It should be `Union[str, Any]` as that is what returns `multihash_as_hex`

File: topo_processor/util/checksum.py [hunk 2]
```diff
@@ -7,7 +8,7 @@
 CHUNK_SIZE = 1024 * 1024  # 1MB
 
 
-def multihash_as_hex(path: str) -> str:
```
Comment (l0b0) [inline topo_processor/util/checksum.py:10]: `Union[str, Any]` === `Any`

**Pull request merged**